flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/decorators/InitJobManagerDecorator.java

```diff
@@ -60,14 +60,14 @@ public FlinkPod decorateFlinkPod(FlinkPod flinkPod) {
                 new PodBuilder(flinkPod.getPod())
                         .withApiVersion(API_VERSION)
                         .editOrNewMetadata()
-                        .withLabels(kubernetesJobManagerParameters.getLabels())
-                        .withAnnotations(kubernetesJobManagerParameters.getAnnotations())
+                        .addToLabels(kubernetesJobManagerParameters.getLabels())
+                        .addToAnnotations(kubernetesJobManagerParameters.getAnnotations())
                         .endMetadata()
                         .editOrNewSpec()
                         .withServiceAccountName(kubernetesJobManagerParameters.getServiceAccount())
```

Comment:
Does it make sense to only override the service account name if an explicit service account name has been configured?

Maybe the following precedence could work:

1. Service account specified via in flink-conf.yaml
2. Service account specified in pod template
3. Default value of `KUBERNETES_SERVICE_ACCOUNT`

Reply:
When I design the pod template at the very begging, I have a simple rule that pod template is used to configure the advanced features(e.g. init container, sidecar container, volumes, etc) which we have not supported via config options. So if a Kubernetes feature has a corresponding config option, the users should use it. If it is not explicitly configured, the default value will take effect. This will force the users to make the pod template as simple as possible.

This is just my idea. Your suggestion about the precedence also makes sense. I will have a try on this.

Reply:
I am not sure whether users will find it very ergonomic to look up which options they must configure via config options and which others they can configure via the template. I could imagine that most K8s user will mostly use the template as it is most natural to do it.

I think we should define how the template, explicit Flink configurations, their default values and system values should interact. 

I can imagine that we have 1) some values which Flink will overwrite. 2) Then there are settings where Flink will merge its values with the user provided values. 3) And last but not least there are values which overwrite Flink settings. 

For 2) and 3) we should define how the user defined values are resolved given that there can be a template setting, an explicitly configured config option and a default value if it has not been set. Moreover, we should define if settings are merged in which order they are merged.